app/controllers/unattended_controller.rb

```diff
@@ -11,14 +11,16 @@ class UnattendedController < ApplicationController
   # We want to find out our requesting host
   before_action :get_host_details, :except => [:hostgroup_template, :built, :failed]
   before_action :get_built_host_details, :only => [:built, :failed]
-  before_action :allowed_to_install?, :except => :hostgroup_template
+  before_action :allowed_to_install?, :except => :hostgroup_template, :unless => -> { ipxe_request? }
   before_action :handle_realm, :if => Proc.new { params[:kind] == 'provision' }
   # all of our requests should be returned in text/plain
   after_action :set_content_type
 
   # maximum size of built/failed request body accepted to prevent DoS (in bytes)
   MAX_BUILT_BODY = 65535
 
+  IPXE_TEMPLATE_PARAMETER = 'ipxe_template'
```

Comment:
Why this is a constant? I'd go with inline string and let the compiler to optimize this.